client/my-sites/checkout/checkout/web-payment-box.jsx

```diff
@@ -0,0 +1,517 @@
+							.then( paymentResponse => {
+								const { payerName, details } = paymentResponse;
+								const { token } = details;
+
+								if ( 'EC_v1' !== token.paymentData.version ) {
+									return; // Not supported yet.
+								}
+
+								this.props.transaction.payment = newCardPayment( {
+									tokenized_payment_data: token.paymentData,
+									name: payerName,
+									country: this.state.country,
+									'postal-code': this.state.postalCode,
+									card_brand: token.paymentMethod.network,
+								} );
+
+								setPayment( this.props.transaction.payment );
+
+								this.props.onSubmit( event );
+
+								paymentResponse.complete( 'success' );
+							} )
+							.catch( error => {
+								debug( 'show error' );
+								debug( error );
+							} );
+					} catch ( e ) {
+						debug( 'show catch' );
+						debug( e );
+					}
+				}
+				break;
+
+			case WEB_PAYMENT_BASIC_CARD_METHOD:
+				{
+					const paymentRequest = this.getPaymentRequestForBasicCard();
+
+					try {
+						paymentRequest
+							.show()
+							.then( paymentResponse => {
+								const { details } = paymentResponse;
+								const { billingAddress } = details;
+
+								const cardRawDetails = {
+									number: details.cardNumber,
+									cvv: details.cardSecurityCode,
+									'expiration-date': details.expiryMonth + '/' + details.expiryYear.substr( 2 ),
+									name: details.cardholderName,
+									country: billingAddress.country,
+									'postal-code': billingAddress.postalCode,
+								};
+
+								this.props.transaction.payment = newCardPayment( cardRawDetails );
+
+								paymentResponse.complete();
+
+								return this.props.transaction.payment;
+							} )
+							.then( transactionPayment => {
+								setPayment( transactionPayment );
+								this.props.onSubmit( event );
+							} )
+							.catch( error => {
+								debug( error );
+							} );
+					} catch ( e ) {
+						debug( e );
+					}
+				}
+				break;
+
+			default:
+				debug( 'Unknown or unhandled payment method `' + paymentMethod + '`.' );
+		}
+	};
+
+	/**
+	 * @param {string} key    Should only be `country`.
+	 * @param {string} value  Should only be a country name.
+	 */
+	updateSelectedCountry = ( key, value ) => {
+		if ( 'country' === key ) {
+			this.setState( { country: value } );
+		}
+	};
+
+	/**
+	 * @param {object} event  Event object.
+	 */
+	updateSelectedPostalCode = event => {
+		const { name: key, value } = event.target;
+
+		if ( 'postal-code' === key ) {
+			this.setState( { postalCode: value } );
+		}
+	};
+
+	render() {
+		const paymentMethod = detectWebPaymentMethod();
+		const { cart, translate, countriesList } = this.props;
+
+		if ( ! paymentMethod ) {
+			return <></>;
+		}
+
+		const buttonState = this.buttonState();
+		const buttonDisabled = buttonState.disabled || ! this.state.country || ! this.state.postalCode;
+		let introductionText;
+		let button;
+
+		switch ( paymentMethod ) {
+			case WEB_PAYMENT_APPLE_PAY_METHOD:
```

Comment:
Instead of this large switch statement, what if we did something like `const introductionText = this.getIntroductionText()` and `const button = `this.getFormButton()`?

Reply:
Since the introduction text has been removed (as suggested by @rralian), I consider having a `getFormButton` function only useless :-).